authentication/serializers.py

```diff
@@ -93,18 +98,18 @@ class Meta(UserSerializer.Meta):
 
 class CustomTokenCreateSerializer(TokenCreateSerializer):
     def validate(self, attrs):
-        password = attrs.get("password")
-        params = {
-            settings.LOGIN_FIELD: attrs.get(settings.LOGIN_FIELD).lower()
-        }
+        try:
+            return self.validate_for_rate(attrs)
+        except RateLimitException:
+            self.fail("inactive_account")
 
-        self.user = authenticate(
-            request=self.context.get("request"), **params, password=password
+    @RateLimitDecorator(
+        calls=custom_settings.ATTEMPTS_FOR_LOGIN,
+        period=custom_settings.DELAY_FOR_LOGIN,
+    )
+    def validate_for_rate(self, attrs):
+        email = attrs.get(settings.LOGIN_FIELD).lower()
+        new_attr = OrderedDict(
+            [("password", attrs.get("password")), ("email", email)]
```

Comment:
Is there any reasons to use OrderedDict here with two keys?

Reply:
Thank you for feedback. I've deleted OrderedDict